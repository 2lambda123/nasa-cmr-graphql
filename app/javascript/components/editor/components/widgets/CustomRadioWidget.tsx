import React from 'react'
import { WidgetProps } from '@rjsf/utils'
import './Widget.css'

interface CustomRadioWidgetProps extends WidgetProps {
  uiSchema: object,
  onChange: (value: boolean) => void,
}
class CustomRadioWidget extends React.Component<CustomRadioWidgetProps, never> {
  render() {
    const {
      id, value, onChange, uiSchema
    } = this.props
    const valueRequired = uiSchema ? uiSchema['ui:title'] : null
    const componentId = 'custom-radio-widget'
    return (
      <div className="custom-radio-widget" data-testid={`${componentId}`}>
        <div data-testid={`${componentId}--value-required`}>{valueRequired}</div>
        <input
          type="radio"
          id={id}
          checked={value === true}
          onChange={() => onChange(true)}
          data-testid={`${componentId}--true`}
        />
        <label htmlFor={id}>True</label>
        <br />
        <input
          type="radio"
          id={id}
          checked={value === false}
          onChange={() => onChange(false)}
          data-testid={`${componentId}--false`}
        />
        <label htmlFor={id}>False</label>
      </div>
    )
  }
}

export default CustomRadioWidget
