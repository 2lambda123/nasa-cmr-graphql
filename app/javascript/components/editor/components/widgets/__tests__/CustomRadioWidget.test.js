import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import CustomRadioWidget from '../CustomRadioWidget'

describe('CustomRadiowidget', () => {
  it('renders the component with initial values', async () => {
    const props = {
      value: true,
      onChange: jest.fn(),
      uiSchema: { 'ui:title': 'Title' }
    }
    const { container } = render(<CustomRadioWidget {...props} />)
    const labelElement = screen.getByTestId('custom-radio-widget--value-required')
    const trueRadioButton = screen.getByTestId('custom-radio-widget--true')
    const falseRadioButton = screen.getByTestId('custom-radio-widget--false')

    expect(labelElement.textContent).toBe('Title')
    expect(trueRadioButton.checked).toBe(true)
    expect(falseRadioButton.checked).toBe(false)

    expect(container).toMatchSnapshot()
  })

  it('should call the onChange function with true when "True" radio button is clicked', () => {
    const props = {
      value: false,
      onChange: jest.fn(),
      uiSchema: null
    }
    const { container } = render(<CustomRadioWidget {...props} />)
    const trueRadioButton = screen.getByTestId('custom-radio-widget--true')

    fireEvent.click(trueRadioButton)
    expect(props.onChange).toHaveBeenCalledWith(true)

    expect(container).toMatchSnapshot()
  })

  it('should call the onChange function with false when "False" radio button is clicked', () => {
    const props = {
      value: true,
      onChange: jest.fn(),
      uiSchema: null
    }
    const { container } = render(<CustomRadioWidget {...props} />)
    const falseRadioButton = screen.getByTestId('custom-radio-widget--false')

    fireEvent.click(falseRadioButton)
    expect(props.onChange).toHaveBeenCalledWith(false)

    expect(container).toMatchSnapshot()
  })
})
