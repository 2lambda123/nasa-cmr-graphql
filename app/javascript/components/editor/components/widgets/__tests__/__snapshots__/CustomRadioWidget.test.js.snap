// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadiowidget renders the component with initial values 1`] = `
<div>
  <div
    class="custom-radio-widget"
    data-testid="custom-radio-widget"
  >
    <div
      data-testid="custom-radio-widget--value-required"
    >
      Title
    </div>
    <input
      checked=""
      data-testid="custom-radio-widget--true"
      type="radio"
    />
    <label>
      True
    </label>
    <br />
    <input
      data-testid="custom-radio-widget--false"
      type="radio"
    />
    <label>
      False
    </label>
  </div>
</div>
`;

exports[`CustomRadiowidget should call the onChange function with false when "False" radio button is clicked 1`] = `
<div>
  <div
    class="custom-radio-widget"
    data-testid="custom-radio-widget"
  >
    <div
      data-testid="custom-radio-widget--value-required"
    />
    <input
      checked=""
      data-testid="custom-radio-widget--true"
      type="radio"
    />
    <label>
      True
    </label>
    <br />
    <input
      data-testid="custom-radio-widget--false"
      type="radio"
    />
    <label>
      False
    </label>
  </div>
</div>
`;

exports[`CustomRadiowidget should call the onChange function with true when "True" radio button is clicked 1`] = `
<div>
  <div
    class="custom-radio-widget"
    data-testid="custom-radio-widget"
  >
    <div
      data-testid="custom-radio-widget--value-required"
    />
    <input
      data-testid="custom-radio-widget--true"
      type="radio"
    />
    <label>
      True
    </label>
    <br />
    <input
      checked=""
      data-testid="custom-radio-widget--false"
      type="radio"
    />
    <label>
      False
    </label>
  </div>
</div>
`;
